creator-node/src/routes/nodeSync.js

```diff
@@ -308,19 +308,21 @@ async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint, dbOnlySync
           transaction
         })
         const fetchedLatestBlockNumber = fetchedCNodeUser.latestBlockNumber
+        const fetchedLatestClockVal = fetchedCNodeUser.clock
 
         // Delete any previously stored data for cnodeUser in reverse table dependency order (cannot be parallelized).
         if (cnodeUser) {
           // Ensure imported data has higher blocknumber than already stored.
-          // TODO - replace this check with a clock check (!!!)
           const latestBlockNumber = cnodeUser.latestBlockNumber
+          const latestClockValue = cnodeUser.clock
 
           if (!dbOnlySync) {
             if ((fetchedLatestBlockNumber === -1 && latestBlockNumber !== -1) ||
-              (fetchedLatestBlockNumber !== -1 && fetchedLatestBlockNumber <= latestBlockNumber)
+              (latestClockValue >= fetchedLatestClockVal)
             ) {
               throw new Error(`Imported data is outdated, will not sync. Imported latestBlockNumber \
-                ${fetchedLatestBlockNumber} Self latestBlockNumber ${latestBlockNumber}`)
+                ${fetchedLatestBlockNumber} Self latestBlockNumber ${latestBlockNumber}. \
+                fetched latestClockVal: ${fetchedLatestClockVal}, self latestClockVal: ${latestClockValue}`)
```

Comment:
if they are equal, just return a success response, and log that no sync was necessary